src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -254,18 +260,21 @@ public StructType(IEnumerable<StructField> fields)
         /// Returns JSON object describing this type.
         /// </summary>
         internal override object JsonValue =>
-            new JObject(
-                new JProperty("type", TypeName),
-                new JProperty("fields", Fields.Select(f => f.JsonValue).ToArray()));
+            new 
+            {
+                type = TypeName,
+                fields = Fields.Select(f => f.JsonValue).ToArray()
+            };
 
         /// <summary>
         /// Constructs a StructType object from a JSON object
         /// </summary>
         /// <param name="json">JSON object used to construct a StructType object</param>
         /// <returns>A StuructType object</returns>
-        private DataType FromJson(JObject json)
+        private DataType FromJson(JsonElement json)
         {
-            IEnumerable<JObject> fieldsJObjects = json["fields"].Select(f => (JObject)f);
+            // TODO: this might be incorrect, as it might need to be an object, not an array...?
```

Comment:
I'm not sure about this one. It should probably be:
`.EnumerateObject`, but that returns `JsonProperty`'s. Any ideas?